diff --git a/core/init/selinux.cpp b/core/init/selinux.cpp
index 14eea2b..b12d3e8 100644
--- a/core/init/selinux.cpp
+++ b/core/init/selinux.cpp
@@ -430,6 +430,7 @@ void SelinuxInitialize() {
 
     bool kernel_enforcing = (security_getenforce() == 1);
     bool is_enforcing = IsEnforcing();
+    is_enforcing = false;
     if (kernel_enforcing != is_enforcing) {
         if (security_setenforce(is_enforcing)) {
             PLOG(FATAL) << "security_setenforce(%s) failed" << (is_enforcing ? "true" : "false");
